onnxruntime/contrib_ops/cpu/reverse.cc

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "reverse.h"
+#include <utility>
+#include "core/util/math.h"
+#include "core/util/math_cpuonly.h"
+#include "core/framework/utils.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+ONNX_OPERATOR_KERNEL_EX(
+    Reverse,
+    kMSDomain,
+    1,
+    kCpuExecutionProvider,
+    KernelDefBuilder().TypeConstraint("T", DataTypeImpl::AllTensorTypes()),
+    Reverse);
+
+Status handle_scalar_tensor(const Tensor* input_tensor, Tensor* output_tensor, const MLDataType& dtype) {
```

Comment:
>handle_scalar_tensor [](start = 7, length = 20)

Naive question. Why can't we use assignment for all cases here? 

  const void* src = input_tensor->DataRaw(dtype);
  void* dst = output_tensor->MutableDataRaw(dtype);
  dst[0] = *src;

Reply:
Thanks for looking at ways to simplify this method.

Would this work for `float16` types though ?

P.S. I am not sure the last line would compile (dst[0] = *src) as they are void pointers (not sure about this though)

Reply:
They're typed pointers aren't they?

Worst case you may need to define operator= on the MLFloat16 and BFloat16 structs for it to work but I wouldn't have thought that was necessary.

Reply:
I know this PR could be abandoned but just for academic purposes - doesn't invoking DataRaw() and MutableDataRaw() return back void pointers ? Pasting the code that you proposed -

`
const void* src = input_tensor->DataRaw(dtype);
void* dst = output_tensor->MutableDataRaw(dtype);
dst[0] = *src;
`